Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/OnboardingGenerateKeyViewController.swift

```diff
@@ -61,12 +61,32 @@ class OnboardingGenerateKeyViewController: UITableViewController {
         vc.placeholder = "Enter name"
         vc.address = privateKey.address
         vc.completion = { [unowned self, vc] name in
-            guard PrivateKeyController.importKey(privateKey, name: name, isDrivedFromSeedPhrase: true) else { return }
+            guard PrivateKeyController.importKey(privateKey, name: name, isDrivedFromSeedPhrase: true),
+                  let keyInfo = try? KeyInfo.keys(addresses: [privateKey.address]).first,
+                  let sceneDelegate = UIApplication.shared.connectedScenes.first?.delegate as? SceneDelegate else {
+                return
+            }
+
             if App.shared.auth.isPasscodeSet {
-                App.shared.snackbar.show(message: "Owner key successfully imported")
-                vc.dismiss(animated: true, completion: nil)
+                vc.dismiss(animated: false) {
+                    App.shared.snackbar.show(message: "Owner key successfully imported")
+                    sceneDelegate.show(OwnerKeyDetailsViewController(keyInfo: keyInfo))
+                }
             } else {
-                let createPasscodeViewController = CreatePasscodeViewController()
+                let createPasscodeViewController = CreatePasscodeViewController {
+                    App.shared.snackbar.show(message: "Owner key successfully imported")
```

Comment:
Snackbar message: `The key successfully created. Add it to the safe using the desktop app and then restart mobile app.`

Reply:
ok